helper/impls/helper-impl-14/src/main/java/com/linkedin/avroutil1/compatibility/avro14/Avro14Adapter.java

```diff
@@ -106,6 +109,11 @@ public BinaryDecoder newBinaryDecoder(ObjectInput in) {
     return newBinaryDecoder(new ObjectInputToInputStreamAdapter(in), false, null);
   }
 
+  @Override
+  public BinaryDecoder configureBinaryDecoder(byte[] bytes, int offset, int length, BinaryDecoder reuse) {
+    return configurer.configureBinaryDecoder(bytes, offset, length, reuse);
```

Comment:
Why do we need a Configurer? Why not do it inline in this class like in `newBinaryEncoder(OutputStream out, boolean buffered, BinaryEncoder reuse)`?

BTW, I think we need to have the same checks as in the above function to deal with null `reuse` instances and also to check that the type is one we expect...

Reply:
because the method `newBinaryEncoder ` is using is public, but the method we are using is package-private and is not accessible in linkedin package.

Reply:
Ooooh I see. Thanks. Please add a JavaDoc comment on the Configurer classes to point that out.